src/System Application/App/Guided Experience/src/Guided Experience/GuidedExperienceImpl.Codeunit.al

```diff
@@ -507,21 +507,54 @@ codeunit 1991 "Guided Experience Impl."
 
             if GuidedExperienceItem.FindSet() then begin
                 // this part is necessary to include the assisted setup group as a header on the page
-                GuidedExperienceItemTemp.Init();
-                GuidedExperienceItemTemp.Code := Format(GroupId);
-                GuidedExperienceItemTemp."Object ID to Run" := GroupId;
-                GuidedExperienceItemTemp.Title := Format(GroupValue);
-                GuidedExperienceItemTemp."Assisted Setup Group" := GroupValue;
-                GuidedExperienceItemTemp.Insert();
+                TempGuidedExperienceItem.Init();
+                TempGuidedExperienceItem.Code := Format(GroupId);
+                TempGuidedExperienceItem."Object ID to Run" := GroupId;
+                TempGuidedExperienceItem.Title := Format(GroupValue);
+                TempGuidedExperienceItem."Assisted Setup Group" := GroupValue;
+                TempGuidedExperienceItem.Insert();
 
                 GroupId -= 1;
 
-                InsertGuidedExperienceItemsInTempVar(GuidedExperienceItem, GuidedExperienceItemTemp);
+                InsertGuidedExperienceItemsInTempVar(GuidedExperienceItem, TempGuidedExperienceItem);
             end;
         end;
     end;
 
-    procedure GetContentForSetupPage(var GuidedExperienceItemTemp: Record "Guided Experience Item" temporary; GuidedExperienceType: Enum "Guided Experience Type")
+    procedure GetContentForManualSetup(var TempGuidedExperienceItem: Record "Guided Experience Item" temporary)
+    var
+        GuidedExperienceItem: Record "Guided Experience Item";
+        GroupValue: Enum "Manual Setup Category";
+        GroupId: Integer;
+        i: Integer;
+    begin
+        GuidedExperienceItem.SetCurrentKey("Guided Experience Type", "Object Type to Run", "Object ID to Run", "Manual Setup Category", Link, Version);
+        GuidedExperienceItem.SetRange("Guided Experience Type", GuidedExperienceItem."Guided Experience Type"::"Manual Setup");
+        GuidedExperienceItem.SetAscending(Version, false);
+
+        GroupId := -1;
+        foreach i in "Manual Setup Category".Ordinals() do begin
+            GroupValue := "Manual Setup Category".FromInteger(i);
+            GuidedExperienceItem.SetRange("Manual Setup Category", GroupValue);
+
+            if GuidedExperienceItem.FindSet() then begin
```

Comment:
Consider adding Read Isolation so we do not lock the records incidentally.

Reply:
Smart to consider this, true, added.